packages/core/src/Page/index.ts

```diff
@@ -371,7 +373,7 @@ export class Page {
       ...this.pageConfig.frontmatterOverride,
     };
 
-    this.title = this.title || this.frontmatter.title || '';
+    this.title = this.title || this.frontmatter.title || TITLE_DEFAULT;
```

Comment:
Can be changed to ```this.pageConfig.src``` to get the name of the file.
Can add some cleaning up to remove the extension and maybe convert the name from camelCase / snake_case to Capitalised Words

Reply:
Yep, I suppose having the name of the file as the stand-in title is more conventional :grin: 

I am wondering if removing the extension + name conversion is strictly necessary though. If a new user deploys a site that they have built & the title of the site shown on the browser tab is something like "index.md", then they would know that something needs to be changed (and will subsequently look into it... hopefully), but it won't be a stretch for them to assume that it is taking it directly from the file name. If it is instead something cleaner like "Index", then they might be confused on where it came from, and also, it might be  difficult to capture all the edge cases when it comes to cleaning the file name :sweat_smile:

Reply:
> This can be documented so maybe not so much of an issue?

Agreed! My initial thought process was that having three different ways to declare the `title` could be messy, especially if the final `title` is not even the same as the file name. But I suppose it would then be up to the user to read the documentation & up to us to write clear documentation :smile: 

> It could be something simple like [remove file extension](https://stackoverflow.com/questions/4250364/how-to-trim-a-file-extension-from-a-string-in-javascript) and splitting the words. If there are edge cases, users can fall back on using the frontmatter.title.

Yes, I suppose frontmatter could be the backup option for cases that cannot be covered - and the documentation should reflect this clearer (I'll update the documentation in this PR, of course). 

The major change from this update would then be that now the primary way to declare titles would be through the file name, which can be overriden to have custom titles via the frontmatter, which can again be overriden through `site.json`.